ui/src/app/categories/page.tsx

```diff
@@ -21,58 +21,79 @@ export interface Category {
     createdAt: string;
 }
 
-export default async function Categories() {
-    const cookieStore = await cookies();
-    const response = await fetch(`${process.env.NEXT_PUBLIC_API_ENDPOINT}/category/${cookieStore.get('email')?.value}`);
-    const data = await response.json()
+export default function Categories() {
+    const [categories, setCategories] = useState<Category[]>([]);
+    const [userEmail, setUserEmail] = useState("");
 
-    return (
-        <Box sx={{ padding: '16px 16px 0' }}>
-            <Typography
-                variant="h4"
-                gutterBottom
-                sx={{
-                    color: 'text.primary',
-                    fontWeight: 'bold',
-                }}
-                >
-                Categories
-            </Typography>
+    const fetchCategories = async (email: string) => {
```

Comment:
For now we can leave liek this, in the future we need to create a reusable method that can used without needing each time to provide base url.